apps/api/src/app/integrations/usecases/get-webhook-support-status/get-webhook-support-status.usecase.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { IntegrationEntity, IntegrationRepository } from '@novu/dal';
+import { IEmailProvider, ISmsProvider } from '@novu/stateless';
+import { MailFactory, SmsFactory, ISmsHandler, IMailHandler } from '@novu/application-generic';
+import { ChannelTypeEnum } from '@novu/shared';
+import { GetWebhookSupportStatusCommand } from './get-webhook-support-status.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+@Injectable()
+export class GetWebhookSupportStatus {
+  public readonly mailFactory = new MailFactory();
+  public readonly smsFactory = new SmsFactory();
+  private provider: IEmailProvider | ISmsProvider;
+
+  constructor(private integrationRepository: IntegrationRepository) {}
+
+  async execute(command: GetWebhookSupportStatusCommand): Promise<boolean> {
+    const { providerId } = command;
+
+    const integration: IntegrationEntity = await this.getIntegration(command);
+    const { channel } = integration;
+
+    if (!integration) {
+      throw new NotFoundException(`Integration for ${providerId} was not found`);
+    }
+
+    if (![ChannelTypeEnum.EMAIL, ChannelTypeEnum.SMS].includes(channel)) {
+      throw new ApiException(`Webhook for ${providerId}-${channel} is not supported yet`);
+    }
+
+    if (channel === 'sms' || channel === 'email') {
+      this.createProvider(integration, providerId, channel);
+    }
+
+    if (!this.provider.getMessageId || !this.provider.parseEventBody) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private async getIntegration(command: GetWebhookSupportStatusCommand): Promise<IntegrationEntity> {
+    return await this.integrationRepository.findOne({
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+      providerId: command.providerId,
+    });
+  }
+
+  private getHandler(integration: IntegrationEntity, channel: 'email' | 'sms'): ISmsHandler | IMailHandler {
+    switch (channel) {
+      case 'sms':
+        return this.smsFactory.getHandler(integration);
+      default:
+        return this.mailFactory.getHandler(integration);
+    }
+  }
+
+  private createProvider(integration: IntegrationEntity, providerId: string, channel: 'email' | 'sms') {
+    const handler = this.getHandler(integration, channel);
+    if (!handler) {
+      throw new NotFoundException(`Handler for integration of ${providerId} was not found`);
+    }
+    handler.buildProvider({});
+
+    this.provider = handler.getProvider();
+  }
```

Comment:
I think I didn't pay attention properly in my previous review because now I noticed we can simplify the arguments of the methods.
`createProvider` can only receive `integration` as there it carries the providerId and the channel properties.
And `getHandler` only needs `integration` property as well.
```suggestion
  private getHandler(integration: IntegrationEntity): ISmsHandler | IMailHandler {
    switch (integration.channel) {
      case 'sms':
        return this.smsFactory.getHandler(integration);
      default:
        return this.mailFactory.getHandler(integration);
    }
  }

  private createProvider(integration: IntegrationEntity) {
    const handler = this.getHandler(integration);
    if (!handler) {
      throw new NotFoundException(`Handler for integration of ${integration.providerId} was not found`);
    }
    handler.buildProvider({});

    this.provider = handler.getProvider();
  }
```

Reply:
Ah, right it can be simplified